app/javascript/src/components/Dashboard/Hours.jsx

```diff
@@ -0,0 +1,326 @@
+       </Card.Description>
+      </Card.Content>
+      <Card.Content >
+        <Grid.Column>
+          <div className="badge badge-info" >
+            HourRank: {this.hourRank(props.user)}
+          </div>
+       </Grid.Column>
+       <Grid.Column className = {this.coloring(props.user, "night")}>
+         NightRank: {this.nightRank(props.user)}
+       </Grid.Column>
+      </Card.Content>
+    </Card>
+  }
+
+// Trying to move the dropDown options out of the render component, but dropDown fucntion  is buggy
+/**
+dropDown () {
+  const lengthOptions = [{
+      key: 'Week',
+      text: 'Week',
+      value: 'Week',
+      content: 'Week',
+    },
+    {
+      key: 'Total',
+      text: 'Total',
+      value: 'Total',
+      content: 'Total',
+    },
+  ];
+
+  const rankOptions = [{
+      key: 'Hour',
+      text: 'Hour',
+      value: 'Hour',
+      content: 'Hour',
+    },
+    {
+      key: 'Night',
+      text: 'Night',
+      value: 'Night',
+      content: 'Night',
+    },
+  ];
+
+  return <div  className="sorting-dropdowns" style={{
+      flex: 1,
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      display: 'inline-block',
+    }}
+  >
+    <div style={{display: 'inline'}}>
+      <div className="week-dropdown">
+        <div className="dropdown">
+          View by {' '}
+          <Dropdown
+            labeled
+            icon='left dropdown'
+            icon="calendar"
+            inline
+            header='Adjust time span'
+            options={lengthOptions}
+            defaultValue={lengthOptions[0].value}
+            onChange={(e, value) => {
+              const Totalv = value.value === 'Week' ? false : true;
+              console.log( Totalv, value);
+              this.setState({Total:Totalv});
+            }}
+          />
+       </div>
+      </div>
+      <br />
+       <div className="rank-dropdown">
+         <div className="dropdown">
+           Rank by {' '}
+           <Dropdown
+             labeled
+             icon="trophy"
+             inline
+             header='Adjust time span'
+             options={rankOptions}
+             defaultValue={rankOptions[0].value}
+             onChange={ (e, value) => this.handleNightorHours(e, value.value) }
+           />
+        </div>
+       </div>
+    </div>
+  </div>
+}
+*/
+
+
+  render () {
+    const lengthOptions = [{
+        key: 'Week',
+        text: 'Week',
+        value: 'Week',
+        content: 'Week',
+      },
+      {
+        key: 'Total',
+        text: 'Total',
+        value: 'Total',
+        content: 'Total',
+      },
+    ]
+
+    const rankOptions = [{
+        key: 'Hour',
+        text: 'Hour',
+        value: 'Hour',
+        content: 'Hour',
+      },
+      {
+        key: 'Night',
+        text: 'Night',
+        value: 'Night',
+        content: 'Night',
+      },
+    ]
+    const { Total, Hours } = this.state;
+    return (
+    <Card fluid >
```

Comment:
Really long. Consider breaking it into components and importing them

Reply:
Quite difficult to break up into components. Please look at new code to see if this is still an issue